subprojects/docs/src/docs/userguide/lazyConfiguration.adoc

```diff
@@ -15,40 +15,226 @@
 [[lazy_configuration]]
 == Lazy Configuration
 
+As a build grows in complexity, knowing when and where a particular value is configured can become difficult to reason about. Gradle provides several ways to manage this complexity with lazy configuration.
 
 [[sec:lazy_properties]]
 === Lazy properties
 
 [NOTE]
 ====
 
-The provider API is currently <<feature_lifecycle,incubating>>. Please be aware that the DSL and other configuration may change in later Gradle versions.
+The Provider API is currently <<feature_lifecycle,incubating>>. Please be aware that the DSL and other configuration may change in later Gradle versions.
 
 ====
 
-[[sec:creating_new_property_instances]]
-=== Creating new instances
+First, Gradle provides lazy properties, which delay the calculation of a property’s value until it’s absolutely required. This provides two main benefits to build script and plugin authors:
+
+1. Build authors can wire together Gradle models without worrying when a particular property’s value will be known.
+2. Build authors can avoid resource intensive work during the configuration phase, which can have a direct impact on maximum build performance.
+
```

Comment:
For readers it would be helpful if we could spell out some typical situations/use cases where lazy evaluation is helpful e.g. a user wants to map properties in an extension to task properties but the values aren't known until the build script configures them.